SEED-platform/seed
# Feat/better integration

#### Any background context you want to provide?
Initial effort to connect SEED and BETTER via Bsync and Analysis framework.  This PR only tackles the first half of the deliverable:
 - Front end for user inputs for BETTER 
 - property with > 12 months Electricity and Natural Gas meter data converted into Bsync File
 - Analysis creation with the proper payload to BETTER

What this PR does not yet include:
- Actual requests to the BETTER API for building creation and subsequent analysis creation

Because this is likely a large PR I wanted to make sure I was going in the right direction preemptively and resolve any issues as I go.  

#### What's this PR do?
Lays the groundwork for eventual integration with the BETTER API

#### How should this be manually tested?
1. Upload demo properties from the attached file similar to the `bsyncr` demo
2. Match the properties on `PM Property ID' 
3. Upload the PM meter data file also attached
Make sure the 12 months of electricity and natural gas meters have been linked to `123 Location and meters lane`
4. Edit the property to include the `City`, `State`, `Groos Floor Area`, and `Postal Code` 
5. Run the BETTER analysis with arbitrary values (should fail with a message saying it is not connected to BETTER)
Additionally - if you would like to view the BuildingSync document produced from the pipeline you can do so in `../seed/analysis_pipelines/better.py/`. 
#### What are the relevant tickets?
#2657 

#### Supporting Files:
[4 Properties - lat:long and address.xlsx](https://github.com/SEED-platform/seed/files/6555715/4.Properties.-.lat.long.and.address.xlsx)
[PM.Meter.Data.BETTER.input (1).xlsx](https://github.com/SEED-platform/seed/files/6650528/PM.Meter.Data.BETTER.input.1.xlsx)


## UPDATE (06/08)
1. BETTER API calls have been added and analysis results are now served as an HTML file.  The endpoint still has a `@login required` constraint however I have requested they switch this to token auth similar to the other endpoints.    


## UPDATE (06/14)
1. Should now work as expected with the test data
